services/src/main/java/org/keycloak/authorization/util/Permissions.java

```diff
@@ -175,6 +177,8 @@
                         // Later they will be filtered based on any denied scope, if any.
                         // TODO: we could probably provide a configuration option to let users decide whether or not a resource-based permission should grant all scopes associated with the resource.
                         grantedScopes.addAll(permission.getScopes());
+                    } if (resource.isOwnerManagedAccess() && "uma".equals(policy.getType())) {
+                        userManagedPermissions.add(policyResult);
                     }
                     deniedCount--;
                 } else {
```

Comment:
shouldn't be collected also the denied user managed policies?

Reply:
Denied permissions are already collected and considered when calculating the scopes the user is allowed to access. However, user-managed policies that evaluated to a PERMIT override the final decision. For instance, suppose scopes A, B and C were denied for Resource Z. If there is a user managed policy that grants access to scopes A and C, these two scopes will be granted at the end although they were denied by other policies.